hazelcast/src/main/java/com/hazelcast/internal/networking/nonblocking/NonBlockingIOThread.java

```diff
@@ -245,7 +254,7 @@ private void coolDown() {
     }
 
     private void selectLoop() throws IOException {
-        while (!isInterrupted()) {
+        while (!stop) {
```

Comment:
I guess we used `interrupt()` in `shutdown()` to interrupt the `taskQueue.poll()` and finish this loop with a single state. After this change we might not be able to leave the loop, if anything else than `shutdown()` can interrupt this thread (since the `stop` flag will not be set). This might also effect the normal `selectLoop()` methods, not just your experimental `selectNowLoop()`.

Do we need a
```
while (!stop) {
    try {
        ...
    } catch (InterruptedException e) {
        Thread.currentThread.interrupt();
        stop = true; // or break / return
    }
}
```
to handle interruption correctly?